src/map/homunculus.c

```diff
@@ -552,6 +552,12 @@ static int homunculus_gainexp(struct homun_data *hd, unsigned int exp)
 
 	hd->homunculus.exp += exp;
 
+	if (hd->master->state.showexp) {
+		char output[256];
+		sprintf(output, "Homunculus Experience Gained Base:%u (%.2f%%)", exp, ((float)exp / (float)hd->exp_next * (float)100));
```

Comment:
need check hd->exp_next for closer to 0 for avoid division by zero

Reply:
updated